lib/ts/controllers/s-popover.ts

```diff
@@ -315,6 +324,193 @@ namespace Stacks {
             });
         }
     }
+
+    /**
+     * Helper to manually show an s-popover element via external JS
+     * @param element the element the `data-controller="s-popover"` attribute is on
+     */
+    export function showPopover(element: HTMLElement) {
+        const { isPopover, controller } = getPopover(element);
+        if (controller) {
+            controller.show();
+        } else if (isPopover) {
+            element.setAttribute("data-s-popover-auto-show", "true");
+        } else {
+            throw `element does not have data-controller="s-popover"`;
+        }
+    }
+
+    /**
+     * Helper to manually hide an s-popover element via external JS
+     * @param element the element the `data-controller="s-popover"` attribute is on
+     */
+     export function hidePopover(element: Element) {
+        const { isPopover, controller, popover } = getPopover(element);
+
+        if (controller) {
+            controller.hide();
+        } else if (isPopover) {
+            element.removeAttribute("data-s-popover-auto-show");
+            if (popover) {
+                popover.classList.remove("is-visible");
+            }
+        } else {
+            throw `element does not have data-controller="s-popover"`;
+        }
+    }
+
+    /**
+     * Options to use when attaching a popover via `Stacks.attachPopover`.
+     * @see Stacks.attachPopover
+     */
+     export interface PopoverOptions {
+        /**
+         * When true, the `click->s-popover#toggle` action will be attached to the controller element or reference element.
+         */
+       toggleOnClick?: boolean;
+        /**
+         * When set, `data-s-popover-placement` will be set to this value on the controller element.
+         */
+        placement?: string;
+
+        /**
+         * When true, the popover will appear immediately when the controller connects.
+         */
+        autoShow?: boolean;
```

Comment:
How difficult would it be to enable:

``` js
/**
* When set, `data-s-popover-toggle-class` will be set to this value on the controller element.
*/
toggleClass?: string;
```

I gave it a whirl. It didn't work on the first click, but on subsequent clicks, it allowed adding `is-selected` as expected on the originating element.

Reply:
This one is tricky-ish. I left the original behavior intact for when the visible class was on a popover at initialization. In this case, the popover did _not_ have the toggle class added. That _should_ be fixed in #616, where we purposefully introduce the regression along with the new `hide-on-outside-click` interaction settings. I'm punting here purposefully and will double check it on #616.